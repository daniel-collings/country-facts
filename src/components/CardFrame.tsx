import { ReactNode } from 'react'

export default function CardFrame({ children }: { children: ReactNode }) {
  return (
    <div className="bg-base-200 rounded-xl p-8">
      <div className="flex flex-col flex-wrap items-center text-center gap-4">
        {children}
      </div>
    </div>
  )
}
