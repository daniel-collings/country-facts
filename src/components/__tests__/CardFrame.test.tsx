import { describe, expect, test } from 'vitest'
import { render } from '@/utils/testUtils.tsx'
import CardFrame from '@/components/CardFrame.tsx'

describe('CardFrame Component', () => {
  test('should render correctly', () => {
    const { asFragment } = render(
      <CardFrame>
        <h1>Hello World</h1>
      </CardFrame>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
