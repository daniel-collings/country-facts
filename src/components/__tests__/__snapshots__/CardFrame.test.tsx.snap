// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CardFrame Component > should render correctly 1`] = `
<DocumentFragment>
  <div
    class="h-full min-h-screen bg-base-100 p-4 sm:p-8"
    data-theme="light"
  >
    <main
      class="max-w-7xl mx-auto px-0 sm:px-6 lg:px-8"
    >
      <div
        class="bg-base-200 rounded-xl p-8"
      >
        <div
          class="flex flex-col flex-wrap items-center text-center gap-4"
        >
          <h1>
            Hello World
          </h1>
        </div>
      </div>
    </main>
  </div>
</DocumentFragment>
`;
